### ISSUE TITLE
amp-list reset-on-refresh will not hide the load-more button

### ISSUE THREAD

**Issue by tharders** [ampproject/amphtml#24034]
## What's the issue?

When using `amp-list` with load more and `reset-on-refresh` activated the list results will be removed when the src changes, but the load more button is still visible.

When the list content is removed, also the load more button should be hidden

## How do we reproduce the issue?

The following code sample runs in the [playground](https://playground.amp.dev/)
The amp list will show some content and the load more button.
Pressing the button triggers a refresh of the list with a delayed response.
You can see that the content is removed, but the show more button is visible until the new result is available.

```html
<!doctype html>
<html ⚡>
<head>
  <meta charset="utf-8">
  <script async src="https://cdn.ampproject.org/v0.js"></script>
  <script async custom-element="amp-list" src="https://cdn.ampproject.org/v0/amp-list-0.1.js"></script>
  <script async custom-element="amp-bind" src="https://cdn.ampproject.org/v0/amp-bind-0.1.js"></script>
  <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
  <link rel="canonical" href="https://amp.dev/documentation/examples/interactivity-dynamic-content/show_more_button/index.html">
  <title>amp-list refresh</title>
  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
</head>
<body>
  <amp-state id="count">
    <script type="application/json">1</script>
  </amp-state>
  <amp-list
    width="auto" height="300"
    layout="fixed-height"
    src="/documentation/examples/api/photo-stream?items=3&left=3"
    [src]="'/documentation/examples/api/slow-json?delay=2000&rnd='+random()"
    binding="refresh"
    reset-on-refresh
    load-more="manual"
    load-more-bookmark="next">
    <template type="amp-mustache">
        <p>{{title}}</p>
    </template>
  </amp-list>
  <button on="tap:AMP.setState({})">set delayed list url</button>
</body>
</html>
```
## What browsers are affected?

All browsers

## Which AMP version is affected?

This behavior was seen with the latest amp version: 1907301630320

**Comment by tharders** [comment]
There is another problem, which seems strongly related to this one.
When we have a `load-more-failed` element and an error on load more, the error message is shown as expected. But if we change some amp state that triggers a reload of the amp-list, the load more error message is still shown.

This problem can be seen on the [amp.dev staging server when opening the search](https://amp-dev-staging.appspot.com/support/) (icon in top right corner).
Search there for `test error` and click 'load more' twice.
The load more error message should be shown.
When you then change the search text, the error message will not disappear, although I would expect this.

**Comment by tharders** [comment]
@cathyxz This is no blocker for us. We hide the load more button with css and the error should not appear too often anyway.
Since I also do not have too much time I think it would be more effective if you fix this later.
Thanks a lot.

**Comment by cathyxz** [comment]
Btw, the original problem of the see-more button being present can be mitigated if you add a placeholder element. This placeholder element should cover the show-more button while data is loading.
